09-intermediate-rails/10-creating-apis.md

```diff
@@ -98,73 +111,82 @@ Notice in the example above, I used `:ok` instead of the official numeric value
-What if we get params that don't match a pet? What do we do? How should our code change? At the very least, we should make sure that we don't throw an error. Also, we should return a **status code** that indicates to the consumer that we couldn't find any content to match their request. Fortunately, the `204` status code exists for exactly this reason. Let's change our `show` method to:
+## Creating a New Pet
+Now that we can send data via our API, the next step is to consume data sent to us. To demonstrate this, we'll create an endpoint that allows a client to add their own pet to our list.
 
-**Note** that we switch from using the `find` method to the `find_by` method because `find` will return an error before getting to the conditional if the given Pet has not been found.
+### Adding a Create Action
+Questions:
+- What should the URI be for our new endpoint? Why?
+- What HTTP verb should it use? Why?
+- In a web app, we would need both a `new` action and a `create` action. Do we need both for an API?
+
+For this endpoint, we'll be handling `POST` requests to the `/pets` URI, routing them to `pets#create`. Set up the route and add a stub controller action now, and check your work with `rails routes`.
+
+The `create` action will look very similar to how it would in a web app. Rails provides the data sent by the user via `params`, so we'll pull this data out and use it to create a pet. Remember to use strong params to make sure only the fields you want go through.
 
 ```ruby
-def show
-  pet = Pet.find_by(id: params[:id])
+# app/controllers/pets_controller.rb
+def create
+  pet = Pet.new(pet_params)
+  pet.save!
+  render status: :ok, json: {}
+end
 
-  if pet
-    render :json => pet, :status => :ok
-  else
-    render :json => [], :status => :no_content
-  end
+private
+def pet_params
+  params.require(:pet).permit(:name, :age, :human)
 end
 ```
 
-## Activity: Adding Search with your Chair Pair
-1. Write at least one positive (search found pets) and one negative (search didn't find pets) test.
-1. Implement the search feature in the `Pet` model
-1. Create a route and controller action for searching for a pet by name
-1. Make the action return a collection of pets as json
+As before, the big difference from a web app is in what we send back. While in a web app we would probably redirect the client to the resource they created, that doesn't make sense for an API. Instead, we just send back some JSON containing the ID of the newly created pet.
 
-## Resources
-- [`.as_json` documentation](http://api.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html#method-i-as_json)
-- [ActiveModel Serializers](http://railscasts.com/episodes/409-active-model-serializers)
-- [blog post by thoughtbot about serialization](http://robots.thoughtbot.com/better-serialization-less-as-json)
+### Sending POST Data
+While we could use our browser to exercise the `index` and `show` endpoints, sending data to our API is a little trickier. For this, we'll use Postman.
```

Comment:
If we're doing some TDD above, would it make more sense to create the test before trying to execute the code in postman?